### PR TITLE
controller: Run UCC Listener if certification mode is set

### PR DESCRIPTION
**Pull request by morantr**
Currently, the UCC Listener has been initiated on the son_master init function.
This has caused to the UCC Listener to bind to the port from the configuration
file.
The problem is that on some platforms, the port that we are using is used by other
applications, and interfere with them.

Move the UCC Listener initiation to slave_join. It will start only once
if not has been started before.

Signed-off-by: Moran Shoeg <moranx.shoeg@intel.com>

#756

### CODE REVIEW

**Review by tomereli — APPROVED** [review]

**Review by vitalybu — APPROVED** [review]
Approving, although this commit doesn't really solve the described problem.
Now the UCC listener will start later, but can still bind to a used port...
Also, if the UCC listener fails to start, the slave_join process is aborted (unless that's the desired behavior for certification mode).

**Comment by tomereli** [comment]
> Approving, although this commit doesn't really solve the described problem.
> Now the UCC listener will start later, but can still bind to a used port...
> Also, if the UCC listener fails to start, the slave_join process is aborted (unless that's the desired behavior for certification mode).

It will be accompanied with a change in the UGW feed - https://gts-chd.intel.com/projects/SW_WAVE/repos/feed_wlan_6x/pull-requests/210/overview

Edit - the change in the UGW feed is to change `prplmesh.config.certification_mode=0` in the under `files/etc/config/prplmesh`. This means that UGW only is not setting certification mode by default.
The same will be done in the RDKB yocto recipe - @morantr please make sure this happens.

**Comment by tomereli** [comment]
Merged since it passed all tests and part of the controller certification tests. Since it is enough to pass a single test in the testbed to verify this change is OK, and we don't have 2 hours to wait for it, then I merged it.

**Comment by arnout** [comment]
I'm sorry, but this is not OK. As @vitalybu notes, this change doesn't really solve the issue described in the commit message. So either the commit message is wrong, or you're just avoiding the problem and it will bite us again later.

If the problem is indeed a conflicting port, then just change the port number! It is read from the config file so you can easily change it locally, or for all I care you can change it for everybody (slightly more tricky because the config file in the certification setup also has to be updated, but not that big a deal).

**Comment by tomereli** [comment]
> I'm sorry, but this is not OK. As @vitalybu notes, this change doesn't really solve the issue described in the commit message. So either the commit message is wrong, or you're just avoiding the problem and it will bite us again later.
> 

You are absolutely right in one thing - the commit message is wrong. It should have said something like "There is currently a bug that even if certification mode is disabled, the UCC listener is still brought up, which is both a security risk and may cause ports clashing with other critical applications. Therefore, this commit changes the boot sequence of the controller to only start the UCC listener after it has received the platform settings and certification mode is indeed enabled in /etc/config/prplmesh"

> If the problem is indeed a conflicting port, then just change the port number! It is read from the config file so you can easily change it locally, or for all, I care you can change it for everybody (slightly more tricky because the config file in the certification setup also has to be updated, but not that big a deal).

As I explained above, this is not true. The conflicting port is just a symptom of a much more critical issue which is the fact we are running certification mode in a real product. Eventually, we should disable certification mode completely somehow, but this anyway should have been done in a separate PR.

Bottom line - there was a review for this PR by both Vitaly and myself. We both didn't pay enough attention for the commit description, will do in the future.
